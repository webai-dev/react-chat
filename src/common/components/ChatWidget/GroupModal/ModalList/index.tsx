import React, { useContext } from "react";
import { Container } from "@material-ui/core";
import { makeStyles } from "@material-ui/styles";

import TeamListContent from "../../ChatList/TeamList/TeamListContent";
import RoomListContent from "../../ChatList/RoomList/RoomListContent";
import { CompanyContext } from "../../../../hooks/useCompanies";
import { ChatRoomsContext } from "../../../../hooks/useChatRooms";

// Global styles
export const useStyles = makeStyles({
  root: {
    justifyContent: "center",
    position: "relative",
    alignItems: "center",
    height: "570px",
    padding: 0,
    paddingLeft: "65px",
    "& .MuiContainer": {
      overflowY: "auto",
    },
  },
});

interface ModalListProps {
  selectedMembers: number[];
  setSelectedMembers: React.Dispatch<React.SetStateAction<number[]>>;
  connectionStatus: number;
}

/**
 * ModalList component
 * @param {{
 * }} props Properties
 * @description ModalList component consists of a list of ModelListItem components
 */
const ModalList: React.FC<ModalListProps> = ({
  selectedMembers = [],
  setSelectedMembers = () => undefined,
  connectionStatus = 1,
}) => {
  // Styles
  const classes = useStyles();
  const { chatRooms } = useContext(ChatRoomsContext);
  const chatRoom = chatRooms && chatRooms.find((room) => room.groupModalOpened);
  const { selectedCompany } = useContext(CompanyContext);

  return (
    <>
      <Container data-testid="modal-list" className={classes.root}>
        {selectedCompany?.name && selectedCompany.name === "Connections" ? (
          <RoomListContent
            modalItem
            selectedMembers={selectedMembers}
            setSelectedMembers={setSelectedMembers}
            connectionStatus={connectionStatus}
          />
        ) : (
          <TeamListContent
            modalItem
            companyId={chatRoom ? chatRoom.companyId : null}
            selectedMembers={selectedMembers}
            setSelectedMembers={setSelectedMembers}
            connectionStatus={connectionStatus}
          />
        )}
      </Container>
    </>
  );
};

export default ModalList;
