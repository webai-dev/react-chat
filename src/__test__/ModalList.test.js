import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import ModalList from "../common/components/ChatWidget/GroupModal/ModalList";
import renderer from "react-test-renderer";

it("renders without crashing", async () => {
  const { getByTestId } = render(<ModalList />);
  const modalList = getByTestId("modal-list");
  expect(modalList).toBeTruthy();
});

it("matches snapshot", async () => {
  let chatFilter = {
      id: null,
      name: "Connections",
    },
    addMembers = [],
    setAddMembers = () => undefined;
  const tree = renderer
    .create(
      <ModalList
        addMembers={addMembers}
        setAddMembers={setAddMembers}
        chatFilter={chatFilter}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
