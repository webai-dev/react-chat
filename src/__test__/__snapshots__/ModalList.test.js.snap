// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="MuiContainer-root makeStyles-root-9 MuiContainer-maxWidthLg"
  data-testid="modal-list"
>
  <div
    className="makeStyles-root-15"
    data-testid="team-list-content"
  >
    <span />
  </div>
</div>
`;
